app/controllers/session_controller.rb

```diff
@@ -73,6 +73,11 @@ def sso_login
         if SiteSetting.must_approve_users? && !user.approved?
           render text: I18n.t("sso.account_not_approved"), status: 403
           return
+        elsif !user.active?
```

Comment:
how often should we hit them with activiation emails? there should be some limit, no?